### PR TITLE
Implement access request link expiration setting

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1387

Note: only the last commit corresponds to the current task. Previous one will be squashed after merge of this https://github.com/inveniosoftware/invenio-app-rdm/pull/2326

Preview:
Screenshot 2023-08-01 at 10 27 09

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/ShareModal.js [hunk 1]
```diff
@@ -5,59 +5,90 @@
...
-          <LinksTab record={record} accessLinksSearchConfig={accessLinksSearchConfig} />
-        </Tab.Pane>
-      ),
-    },
-  ];
+export class ShareModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      record: this.props.record,
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/ShareModal.js:19]: Can you take into account this linting issue?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                            onChange={({ data, formikProps }) => {
+                              formikProps.form.setFieldValue(
+                                "allow_guest_requests",
+                                data.checked
+                              );
+                            }}
+                          />
+                        </Form.Field>
+                        <label className="helptext mb-0 mt-10">
+                          {i18next.t(
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js:153]: Just a comment about the text:
```diff
- You will get an email asking for approval.
+ When access is requested by someone, you will get an e-mail asking for approval.
```

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                          )}
+                        </label>
+                      </Grid.Column>
+                    </Grid.Row>
+                    <Grid.Row>
+                      <Grid.Column>
+                        <Form.Field>
+                          <TextAreaField
+                            placeholder={i18next.t(
+                              "Optional. Specify conditions under which you approve access."
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js:166]:
```suggestion
                              "Optional. Specify conditions under which you approve access. This message will be visible by any user requesting access in the access request box."
```
I don't know if `access request box` is the right term

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    accept_conditions_text: Yup.string().nullable(),
+    secret_link_expiration: Yup.number(),
+  });
+
+  linkExpirationOptions = [
+    { key: 0, text: i18next.t("Never"), value: 0 },
+    { key: 30, text: i18next.t("In 1 month"), value: 30 },
+    { key: 60, text: i18next.t("In 2 months"), value: 60 },
+    { key: 180, text: i18next.t("In 6 months"), value: 180 },
+    { key: 360, text: i18next.t("In 1 year"), value: 360 },
```
Comment (max-moser) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js:48]: nit: i understand that 360 is the logical next value in the series here, but from a user perspective i would expect 365

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js [hunk 5]
```diff
@@ -80,6 +101,9 @@ export class AccessRequestsTab extends Component {
             allow_guest_requests: record.parent.access.settings.allow_guest_requests,
             accept_conditions_text:
               record.parent.access.settings.accept_conditions_text,
+            secret_link_expiration: this.translateExpirationDate(
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessRequests/AccessRequestsTab.js:104]: What about simply doing:
```suggestion
            secret_link_expiration: record.parent.access.settings.secret_link_expiration || 0;
```

**Review by ntarocco — APPROVED** [review]

**Pull request merged**